From a3545edde9310c874c04ed34d45240a3b6f2cfd0 Mon Sep 17 00:00:00 2001
From: John Galt <johngaltfirstrun@gmail.com>
Date: Fri, 28 Mar 2025 12:32:10 -0400
Subject: [PATCH 4/4] iconloaderlib: src: force monochrome icons on themed
 icons

This has been significantly improved, and we can kill our (now broken)
Launcher3 toggle for this.

Change-Id: I2ce0e077422fd9fa4c734f9ee5bf7aabb30362b8
---
 .../android/launcher3/icons/mono/MonoIconThemeController.kt  | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/iconloaderlib/src/com/android/launcher3/icons/mono/MonoIconThemeController.kt b/iconloaderlib/src/com/android/launcher3/icons/mono/MonoIconThemeController.kt
index 1c73dac..8449d87 100644
--- a/iconloaderlib/src/com/android/launcher3/icons/mono/MonoIconThemeController.kt
+++ b/iconloaderlib/src/com/android/launcher3/icons/mono/MonoIconThemeController.kt
@@ -93,10 +93,7 @@ class MonoIconThemeController(
         if (mono != null) {
             return ClippedMonoDrawable(mono, shapePath, iconScale)
         }
-        if (Flags.forceMonochromeAppIcons() && shouldForceThemeIcon && !isFileDrawable) {
-            return MonochromeIconFactory(info.icon.width).wrap(base, shapePath, iconScale)
-        }
-        return null
+        return MonochromeIconFactory(info.icon.width).wrap(base, shapePath, iconScale)
     }
 
     override fun decode(
-- 
2.51.0

